run-tests.py

```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
+          'public interfaces are clean.')
+
+
+def check_exposed_symbols(_module, _submodule_names, _doc_symbols):
+    """
+    Check _module for any classes and methods not contained in _doc_symbols,
+    and check for any modules not contained in _submodule_names.
+
+    :param _module: the module to check
+    :param _submodule_names: list of submodules exposed by _module
+    :param _doc_symbols: dictionary containing lists of documented classes and
+     functions
+    """
+
+    import inspect
+    exposed_symbols = [x for x in dir(_module) if not x.startswith('_')]
+    symbols = [getattr(_module, exposed_symbols[x]) for x in
+               range(len(exposed_symbols))]
+
+    classes = [x for x in symbols if inspect.isclass(x)]
+    functions = [x for x in symbols if inspect.isfunction(x)]
+
+    # Check for modules: these should match perfectly with _submodule_names
+    exposed_modules = [x for x in symbols if inspect.ismodule(x)]
+    unexpected_modules = [m for m in exposed_modules if
+                          m.__name__ not in _submodule_names]
+
+    if len(unexpected_modules) > 0:
+        print('The following modules are unexpectedly exposed in the public '
+              'interface of %s:' % _module.__name__)
+        for m in sorted(unexpected_modules):
+            print('  unexpected module: ' + m.__name__)
+
+        print('For python modules such as numpy you may need to confine the '
+              'import to the function scope. If you have created a new pints '
+              'submodule, you will need to make %s (doctest) aware of this.'
+              % __file__)
+        print('FAILED')
+        sys.exit(1)
+
+    # Check that all classes are documented
+    undocumented_classes = []
+    for _class in classes:
+        class_name = _module.__name__ + '.' + _class.__name__
+        if class_name not in _doc_symbols['classes']:
+            undocumented_classes.append(class_name)
+
+    if len(undocumented_classes) > 0:
+        print('The following classes do not appear in any RST file:')
+        for m in sorted(undocumented_classes):
+            print('  undocumented class: ' + m)
+        print('FAILED')
+        sys.exit(1)
+
+    # Check that all functions are documented
+    undocumented_functions = []
+    for _funct in functions:
+        funct_name = _module.__name__ + '.' + _funct.__name__
+        if funct_name not in _doc_symbols['functions']:
+            undocumented_functions.append(funct_name)
+
+    if len(undocumented_functions) > 0:
+        print('The following functions do not appear in any RST file:')
+        for m in sorted(undocumented_functions):
+            print('  undocumented function: ' + m)
+        print('FAILED')
+        sys.exit(1)
+
+
+def get_all_documented_symbols():
+    """
+    Recursively traverse docs/source and identify all autoclass and
+    autofunction declarations.
+
+    :return: A dict containing a list of classes and a list of functions
```

Comment:
Same as for `:param:` above!
At the moment we just write "Returns: A dict containing..."

Reply:
Or "Returns a dict..."